document/JS.bs

```diff
@@ -0,0 +1,935 @@
+  required DOMString module;
+  required DOMString name;
+  required ImportExportKind kind;
+};
+
+[LegacyNamespace=WebAssembly, Contructor(BufferSource bytes), Exposed=(Window, Worker)]
+interface Module {
+  static sequence&lt;ModuleExportDescriptor> exports(Module module);
+  static sequence&lt;ModuleImportDescriptor> imports(Module module);
+  static sequence&lt;ArrayBuffer> customSections(Module module, DOMString sectionName);
+};
+</pre>
+
+<!-- Any use for this?
+<div algorithm="name of the export">
+    The <dfn>name of the export</dfn> |export| in the module |module| is defined as follows:
+    1. If |export| is of the form [=𝖿𝗎𝗇𝖼=] |funcidx| and |module| has a [=name section=] with a [=function names subsection=],
+         1. Within <code>[=funcnamesubsec=]</code>, search for an entry <code>[=nameassoc=]</code> of the form |funcidx| <code>name</code>. If such an entry is found, return <code>name</code> decoded as a UTF-8 into a JavaScript String.
+    1. Return the empty String.
+</div>
+-->
+
+<div algorithm>
+    The <dfn method for="Module">exports(module)</dfn> method, when invoked, performs the following steps:
+    1. Let |ast| be |module|.\[[Module]].
+    1. Let |exports| be an empty [=list=].
+    1. For each |export| in |ast|.exports:
+        1. Let |name| be |export|.name as a String.
+        1. Let |kind| be
+            * `"function"` if |export|.desc is of the form 𝖿𝗎𝗇𝖼 funcidx
+            * `"table"` if |export|.desc is of the form 𝗍𝖺𝖻𝗅𝖾 tableidx
+            * `"memory"` if |export|.desc is of the form 𝗆𝖾𝗆    memidx
+            * `"global"` if |export|.desc is of the form 𝗀𝗅𝗈𝖻𝖺𝗅 globalidx
+        1. Let |obj| be ! ObjectCreate(%ObjectPrototype%).
+        1. Perform ! CreateDataProperty(obj, `"name"`, |name|).
+        1. Perform ! CreateDataProperty(obj, `"kind"`, |kind|).
+        1. [=Append=] |obj</obj> to the end of |exports|.
+    1. Return ! CreateArrayFromList(|exports|).
+</div>
+
+<div algorithm>
+    The <dfn method for="Module">imports(module)</dfn> method, when invoked, performs the following steps:
+    1. Let |ast| be |module|.\[[Module]].
+    1. Let |imports| be an empty List.
+    1. For each |imports| in |ast|.imports:
+        1. Let |module| be |import|.module as a String.
+        1. Let |name| be |import|.name as a String.
+        1. Let |kind| be
+            * `"function"` if |export| is of the form 𝖿𝗎𝗇𝖼 functype
+            * `"table"` if |export| is of the form 𝗍𝖺𝖻𝗅𝖾 tabletype
+            * `"memory"` if |export| is of the form 𝗆𝖾𝗆  memtype
+            * `"global"` if |export| is of the form 𝗀𝗅𝗈𝖻𝖺𝗅 globaltype
+        1. Let |obj| be ObjectCreate(%ObjectPrototype%).
+        1. Perform ! CreateDataProperty(obj, `"module"`, |module|).
+        1. Perform ! CreateDataProperty(obj, `"name"`, |name|).
+        1. Perform ! CreateDataProperty(obj, `"kind"`, |kind|).
+        1. Append |obj</obj> to the end of |imports|.
+    1. Return ! CreateArrayFromList(|imports|).
```

Comment:
No need to CreateArrayFromList. This is a Web IDL function, which returns `sequence<>`s, which are lists. The Web IDL layer handles the conversion to JS for you.

Reply:
Done